text/0000-time-improvements.md

```diff
@@ -0,0 +1,330 @@
+- Feature Name: time_improvements
+- Start Date: 2015-09-20
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+
+This RFC proposes several new types and associated APIs for working with times in Rust.
+The primary new types are `ProcessTime`, for working with monotonic time within a single
+process, and `SystemTime`, for working with times across processes on a single system
+(usually internally represented as a number of seconds since an epoch).
+
+# Motivations
+
+The primary motivation of this RFC is to flesh out a larger set of APIs for
+representing instants in time and durations of time.
+
+For various reasons that this RFC will explore, APIs related to time are fairly
+error-prone and have a number of caveats that programmers do not expect.
+
+Rust APIs tend to expose more of these kinds of caveats through their APIs, in
+order to help programmers become aware of and handle edge-cases. At the same
+time, un-ergonomic APIs can work against that goal.
+
+This RFC attempts to balance the desire to expose common footguns and help
+programmers handle edge-cases with a desire to avoid creating so many hoops to
+jump through that the useful caveats get ignored.
+
+At a high level, this RFC covers two concepts related to time:
+
+* Instants, moments in time
+* Durations, an amount of time between two instants
+
+We would like to be able to do some basic operations with these instants:
+
+* Compare two instants
+* Add a time period to an instant
+* Subtract a time period from an instant
+* Compare an instant to "now" to discover time elapsed
+
+However, there are a number of problems that arise when trying to define these
+types and operations.
+
+First of all, with the exception of instants created in the rutime of a single
+process, instants are not monotonic. A simple example of this is that if a
```

Comment:
Ok, so we're talking about the proposed API here? Because both Posix and windows provide functions to get a monotonic system time (`clock_gettime` + `CLOCK_MONOTONIC` or `GetTickCount64()`/`GetTickCount()`).

Might want to clearly delineate where we're talking about actual system restrictions vs restrictions that just occur in the Rust api.

The simple example happens to be true because (generally) one cannot specify the clock used to generate a file timestamp.